webapp/channels/src/components/drafts/drafts.tsx

```diff
@@ -137,42 +141,56 @@ function Drafts({
                 })}
             />
 
-            <Tabs
-                id='draft_tabs'
-                activeKey={activeTab}
-                mountOnEnter={true}
-                unmountOnExit={false}
-                onSelect={handleSwitchTabs}
-            >
-                <Tab
-                    eventKey={0}
-                    title={draftTabHeading}
+            {
+                isScheduledPostEnabled &&
+                <Tabs
+                    id='draft_tabs'
+                    activeKey={activeTab}
+                    mountOnEnter={true}
                     unmountOnExit={false}
-                    tabClassName='drafts_tab'
+                    onSelect={handleSwitchTabs}
                 >
-                    <DraftList
-                        drafts={drafts}
-                        user={user}
-                        displayName={displayName}
-                        draftRemotes={draftRemotes}
-                        status={status}
-                    />
-                </Tab>
-
-                <Tab
-                    eventKey={1}
-                    title={scheduledPostsTabHeading}
-                    unmountOnExit={false}
-                    tabClassName='drafts_tab'
-                >
-                    <ScheduledPostList
-                        scheduledPosts={scheduledPosts || EMPTY_LIST}
-                        user={user}
-                        displayName={displayName}
-                        status={status}
-                    />
-                </Tab>
-            </Tabs>
+                    <Tab
+                        eventKey={0}
+                        title={draftTabHeading}
+                        unmountOnExit={false}
+                        tabClassName='drafts_tab'
+                    >
+                        <DraftList
+                            drafts={drafts}
+                            user={user}
+                            displayName={displayName}
+                            draftRemotes={draftRemotes}
+                            status={status}
+                        />
+                    </Tab>
+
+                    <Tab
+                        eventKey={1}
+                        title={scheduledPostsTabHeading}
+                        unmountOnExit={false}
+                        tabClassName='drafts_tab'
+                    >
+                        <ScheduledPostList
+                            scheduledPosts={scheduledPosts || EMPTY_LIST}
```

Comment:
not sure now about the internals of ScheduledPostList, so if defaulting inside the component leads to unnecessary computations on every render then it makes sense to handle the default value outside the component. Otherwise it is a better practice to deal with no/null/undefined/etc values in the component itself.

Reply:
Defaulting here seems cleaner, but yeah, subjective.